### DISCUSSION TITLE
Shared data Client - user web radio stations not visible

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#1069] [Fixed Issues]
Updated to rAudio 20221014 on shared data server and client machines (other server).

Confirmed NFS services were running on NAS.

Although the shared data client item counts (library page) are correct and the shared data appears to be correct (counts, webradio and img folders), clicking on the web radio icon displays only the standard radio paradise stations.   All other library related items (shared data client) are not affected.

I have also tested disabling server and client shared data, deleting shared data file on NAS, then enabling server and client side shared data, but I get the same behavior.

Also Shared data server side correctly displays user web radio stations.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by ceejayemm** [comment]
@Sagnasty 
20221014 works fine for all my WebRadios (based on rAudio-1 provided SharedData setup as previously discussed). Are you still using a separate NFS server /share ?  When I used this method I had lots of problems with rAudio-1 trying to change permissions on files/folders provided by the NFS server which I always argued it shouldn't do.  Check the permissions of the server providing the share and also what rAudio-1 sees on the RPi, might be a clue there.

Chris

**Reply by rern** [reply]
@Sagnasty,

Please try the UPDATE branch.

**Comment by Sagnasty** [comment]
@rern,

Update solved the issue.

I also tested "losing" the NFS server (Other servers: (Alternative 2)), and confirmed that I can also access the users web radio stations along with users station art after selecting yes to the new dialog box (Thanks very much for adding this ability):
![image](https://user-images.githubusercontent.com/76575081/195988664-9dde5fcb-52d8-4f84-80ec-68fac3dff398.png)

However, while shared data is disabled (use local data), the web radio count still shows "4" (should be 34 in my case).

Jim

**Reply by rern** [reply]
Try enable again:

On enable
- All shared Web Radio data saved to local backup
- Any modifications will be saved to local backup of all clients

On disable
- Local backup restored with latest Web Radio data

Also try - with Library page on every clients
- Move items to rearrange order on any clients
- Bookmark - Add / Remove / Edit
- Set show/hide items

**Comment by Sagnasty** [comment]
I tested this some more today - test results:
Moving / rearranging library items on any client resulted in the same change being mirrored on the other connected clients.
Add / Remove / Edit bookmarks on any client resulted in the same change being mirrored on the other connected clients.
Show/hide library items on any client resulted in the same change being mirrored on the other connected clients.

My issue was that when the client was disabled, I had 34 web radio stations with the correct station art visible on the library page, but the displayed web radio count went back to 4.
![image](https://user-images.githubusercontent.com/76575081/196002750-5f0d98fd-3823-495e-aa7b-25f609872dfc.png)

It then occurred to me that the new shared data function is no longer data coming from a master device (pushed to the shared data location on a NFS server at the users request) and then being made available (read only / pull) to slave devices connected to the NFS shared data location.   Now, all enabled shared data devices are a master with respect to changing shared data and requires a full permission share that is exposed to an external network to work in my case.  

Prior to this new behavior, I was using shared data as a mechanism to export / import shared data with each action being initiated by the user only.   I would connect to the shared data folder / ticked "use data from this device" which would save the master data to the NAS and then I would disable the share (and allow me to save a copy of the data).   Then I had a static version of my master data to share with my other devices.

It would be great to have this simple ability back if possible.   Then it is very easy and predictable to maintain master settings on one device (especially web radio stations and a large music Library) and allow these master settings (shared data) to be pulled on the remaining users devices (RO permissions only) on both a local or remote network with low risk.
